### PR TITLE
Unit & Integration Testing

### PR DESCRIPTION
**Pull request by mariodavid**

### CODE REVIEW

File: content/modules/testing/pages/integration-tests.adoc [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+[source,java,indent=0,title='OrderLineEventListener.java']
+----
+include::example$/ex1/src/main/java/testing/ex1/listener/OrderLineEventListener.java[tags=recalculate-order-amount]
+----
+
+In the integration test, we can now test the `OrderLineEventListener` and the `OrderAmountCalculation` together. The `DataManager` is Jmix central entry point to the persistence layer of the application. During the save operation, it is executing the event listeners, that are registered in the application. We will create an order and an order line and store them in the database via the `DataManager` API. This will trigger the event listener and the order amount will be calculated.
+
+== Dependency Injection in Tests
```
Comment (knstvk) [inline content/modules/testing/pages/integration-tests.adoc:14]: Please add anchors to all headers

File: content/modules/testing/pages/integration-tests.adoc [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+It creates a security context before each test and sets the authenticated user to the admin user. This allows you to write tests that require a security context without having to set up the security context yourself.
+
+[source,java,indent=0,title='CustomerServiceTest.java']
+----
+include::example$/ex1/src/test/java/testing/ex1/customer/CustomerServiceTest.java[tags=authenticated-as-admin]
+----
+
+It is also possible to combine the `AuthenticatedAsAdmin` extension with the `SystemAuthenticator` to execute the test code as a particular user. By annotating the test class the default security context is set to the admin. But within the test case, we can use the `SystemAuthenticator` to execute the code as a particular user. This way, we don't have to manually wrap all test data generation code into by a `SystemAuthenticator` call.
+
+== Overriding behaviour of the application
```
Comment (knstvk) [inline content/modules/testing/pages/integration-tests.adoc]: Title Case

**Review by knstvk — CHANGES_REQUESTED** [review]
Need some cosmetic changes

File: content/modules/testing/pages/integration-tests.adoc (hunk 1 above)
Comment (mariodavid) [reply content/modules/testing/pages/integration-tests.adoc:14]: done

File: content/modules/testing/pages/integration-tests.adoc (hunk 2 above)
Comment (mariodavid) [reply content/modules/testing/pages/integration-tests.adoc]: done

**Pull request merged**